### ISSUE TITLE
npm run check-security fails (Large gzip Denial of Service)

### ISSUE THREAD

**Issue by ooby** [brave/browser-laptop#11168]
### Description
`npm run check-security` fails

### Steps to Reproduce
  1. run `npm run check-security`

**Actual result:**
```bash
$ npm run check-security

> brave@0.21.0 check-security C:\dev\browser-laptop
> nsp check

(+) 1 vulnerabilities found
┌───────────────┬─────────────────────────────────────────────────────────────────┐
│               │ Large gzip Denial of Service                                    │
├───────────────┼─────────────────────────────────────────────────────────────────┤
│ Name          │ superagent                                                      │
├───────────────┼─────────────────────────────────────────────────────────────────┤
│ CVSS          │ 3.7 (Low)                                                       │
├───────────────┼─────────────────────────────────────────────────────────────────┤
│ Installed     │ 3.5.2                                                           │
├───────────────┼─────────────────────────────────────────────────────────────────┤
│ Vulnerable    │ All                                                             │
├───────────────┼─────────────────────────────────────────────────────────────────┤
│ Patched       │ None                                                            │
├───────────────┼─────────────────────────────────────────────────────────────────┤
│ Path          │ brave@0.21.0 > ledger-client@0.9.21 > bitgo@3.5.1 > superagent… │
├───────────────┼─────────────────────────────────────────────────────────────────┤
│ More Info     │ https://nodesecurity.io/advisories/479                          │
└───────────────┴─────────────────────────────────────────────────────────────────┘

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! brave@0.21.0 check-security: `nsp check`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the brave@0.21.0 check-security script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

```

**Expected result:**
success

**Reproduces how often:** [What percentage of the time does it reproduce?]
100%

### Brave Version
0.21.0

### Additional information
[Vulnerability opened 27th Sep. 2017](https://nodesecurity.io/advisories/479)

**Comment by zephinzer** [comment]
same here

**Comment by bsclifton** [comment]
Thanks for the report! Comment from @evq:
> I don't think we're affected by this - an attacker would need to control the URL we're requesting. I don't think we're making use of the `prefetch` / `remote` capabilities of bloodhound in the browser. For ledger-client, the urls requested are always bitgo urls - so an attacker cannot control them.

I'll get a patch up soon

**Comment by bsclifton** [comment]
Looks like there's a few more ☹️  Stay tuned

**Comment by LaurenWags** [comment]
@bsclifton based on https://github.com/brave/browser-laptop/issues/11168#issuecomment-332745318 does that mean this isn't ready for testing?

**Comment by kjozwiak** [comment]
> @bsclifton based on #11168 (comment) does that mean this isn't ready for testing?

It looks like all the issues have been resolved on the `0.19.x` branch:

```
Kamils-MBP:browser-laptop kjozwiak$ git branch
* 0.19.x

Kamils-MBP:browser-laptop kjozwiak$ npm run check-security
> brave@0.19.23 check-security /Users/kjozwiak/projects/browser-laptop
> nsp check
(+) No known vulnerabilities found
```

However, it appears there's still a bunch of issues left on the `master' branch:

> Kamils-MBP:browser-laptop kjozwiak$ git branch
> * master
> 
> Kamils-MBP:browser-laptop kjozwiak$ npm run check-security
> > brave@0.21.0 check-security /Users/kjozwiak/projects/browser-laptop
> > nsp check
> 
> (+) 6 vulnerabilities found
> ┌───────────────┬───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
> │               │ Regular Expression Denial of Service                                                                                              │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ Name          │ mime                                                                                                                              │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ CVSS          │ 7.5 (High)                                                                                                                        │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ Installed     │ 1.3.4                                                                                                                             │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ Vulnerable    │ < 1.4.1 || > 2.0.0 < 2.0.3                                                                                                        │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ Patched       │ >= 1.4.1 < 2.0.0 || >= 2.0.3                                                                                                      │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ Path          │ brave@0.21.0 > ledger-client@0.9.21 > bitgo@3.5.1 > express@4.15.5 > send@0.15.6 > mime@1.3.4                                     │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ More Info     │ https://nodesecurity.io/advisories/535                                                                                            │
> └───────────────┴───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
> ┌───────────────┬───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
> │               │ Regular Expression Denial of Service                                                                                              │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ Name          │ mime                                                                                                                              │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ CVSS          │ 7.5 (High)                                                                                                                        │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ Installed     │ 1.3.4                                                                                                                             │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ Vulnerable    │ < 1.4.1 || > 2.0.0 < 2.0.3                                                                                                        │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ Patched       │ >= 1.4.1 < 2.0.0 || >= 2.0.3                                                                                                      │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ Path          │ brave@0.21.0 > ledger-client@0.9.21 > bitgo@3.5.1 > superagent@3.5.2 > mime@1.3.4                                                 │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ More Info     │ https://nodesecurity.io/advisories/535                                                                                            │
> └───────────────┴───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
> ┌───────────────┬───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
> │               │ Regular Expression Denial of Service                                                                                              │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ Name          │ mime                                                                                                                              │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ CVSS          │ 7.5 (High)                                                                                                                        │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ Installed     │ 1.3.4                                                                                                                             │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ Vulnerable    │ < 1.4.1 || > 2.0.0 < 2.0.3                                                                                                        │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ Patched       │ >= 1.4.1 < 2.0.0 || >= 2.0.3                                                                                                      │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ Path          │ brave@0.21.0 > url-loader@0.5.9 > mime@1.3.4                                                                                      │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ More Info     │ https://nodesecurity.io/advisories/535                                                                                            │
> └───────────────┴───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
> ┌───────────────┬───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
> │               │ Regular Expression Denial of Service                                                                                              │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ Name          │ mime                                                                                                                              │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ CVSS          │ 7.5 (High)                                                                                                                        │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ Installed     │ 1.3.4                                                                                                                             │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ Vulnerable    │ < 1.4.1 || > 2.0.0 < 2.0.3                                                                                                        │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ Patched       │ >= 1.4.1 < 2.0.0 || >= 2.0.3                                                                                                      │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ Path          │ brave@0.21.0 > webtorrent-remote@2.0.2 > webtorrent@0.98.19 > mime@1.3.4                                                          │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ More Info     │ https://nodesecurity.io/advisories/535                                                                                            │
> └───────────────┴───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
> ┌───────────────┬───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
> │               │ Regular Expression Denial of Service                                                                                              │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ Name          │ debug                                                                                                                             │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ CVSS          │ 3.7 (Low)                                                                                                                         │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ Installed     │ 2.2.0                                                                                                                             │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ Vulnerable    │ <= 2.6.8 || >= 3.0.0 <= 3.0.1                                                                                                     │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ Patched       │ >= 2.6.9 < 3.0.0 || >= 3.1.0                                                                                                      │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ Path          │ brave@0.21.0 > ledger-client@0.9.21 > bitgo@3.5.1 > morgan@1.5.3 > debug@2.2.0                                                    │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ More Info     │ https://nodesecurity.io/advisories/534                                                                                            │
> └───────────────┴───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
> ┌───────────────┬───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
> │               │ Regular Expression Denial of Service                                                                                              │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ Name          │ debug                                                                                                                             │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ CVSS          │ 3.7 (Low)                                                                                                                         │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ Installed     │ 2.6.7                                                                                                                             │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ Vulnerable    │ <= 2.6.8 || >= 3.0.0 <= 3.0.1                                                                                                     │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ Patched       │ >= 2.6.9 < 3.0.0 || >= 3.1.0                                                                                                      │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ Path          │ brave@0.21.0 > ledger-client@0.9.21 > bitgo@3.5.1 > body-parser@1.17.2 > debug@2.6.7                                              │
> ├───────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
> │ More Info     │ https://nodesecurity.io/advisories/534                                                                                            │
> └───────────────┴───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
> 
> npm ERR! code ELIFECYCLE
> npm ERR! errno 1
> npm ERR! brave@0.21.0 check-security: `nsp check`
> npm ERR! Exit status 1
> npm ERR!
> npm ERR! Failed at the brave@0.21.0 check-security script.
> npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

**Comment by bsclifton** [comment]
This has now been fixed 😄 
0.19.x https://github.com/brave/browser-laptop/commit/8ae5a2f6730e8ca39f4b13f0dd13f41510a689da (includes discussion)
0.20.x https://github.com/brave/browser-laptop/commit/28e1ea2d7ab5ccf1403edab141567fc6c150c111
master https://github.com/brave/browser-laptop/commit/c5a903cc2672577a41fdf8ad070aec245ecaa367
